fawkes/algorithms/summarization/summarization.py

```diff
@@ -0,0 +1,182 @@
+from nltk.tokenize import sent_tokenize
+import os
+import sys
+import pathlib
+from sklearn.cluster import KMeans
+from gensim.summarization.summarizer import summarize
+from sentence_transformers import SentenceTransformer
+import logging
+import fawkes.constants.logs as logs
+
+
+"""
+        @param{list<string>}: sentences - List of review sentence
+        @param{Integer}: num_clusters - User defined number of clusters
+        @returns{list<list<string>}: clustered_sentences - List of clusters of sentences
+
+        Groups similar sentences into clusters
+"""
+# Reference : https://github.com/UKPLab/sentence-transformers
+
+
+def k_means_classification(sentences, num_clusters):
+
+    # Loading pre-trained model - 'distilbert-base-nli-stsb-mean-tokens'
+    embedder = SentenceTransformer('distilbert-base-nli-stsb-mean-tokens')
+    corpus_embeddings = embedder.encode(sentences)
+
+    # Apply k-means to groups similar sentences into 'num_clusters' clusters
+    clustering_model = KMeans(n_clusters=num_clusters)
+    clustering_model.fit(corpus_embeddings)
+    # assign label to each sentence
+    cluster_assignment = clustering_model.labels_
+
+    # group similar labeled sentences
+    clustered_sentences = {new_list: [] for new_list in range(num_clusters)}
+    for i in range(len(cluster_assignment)):
+        temp_list = clustered_sentences[cluster_assignment[i]]
+        temp_list.append(sentences[i])
+        clustered_sentences[cluster_assignment[i]] = temp_list
+
+    return clustered_sentences
+
+
+"""
+        @param{string}: text - text corpus to summarize
+        @param{integer}: word_count - max number of words in a summary
+        @returns{list<string>}: generated summary
+
+        function to summarize a piece of text
+"""
+
+
+def summarize_text(text, word_count):
+    gen_summary = summarize(text, word_count=word_count).split('\n')
+    # remove all empty reviews from the list
+    gen_summary = [i for i in gen_summary if i]
+    return(gen_summary)
+
+
+"""
+        @param{list<Review>}: reviews - List of reviews class object
+        @returns{list<string>}: processed_sentences - List of processed reviews
+
+        Code to preprocess the data
+            - extract messages from review object
+            - tokenize
+            - remove sentences of inappropriate lengths
+"""
+
+
+def preprocess_review(reviews):
+    processed_sentences = []
+
+    for review in reviews:
+        # Extract a message from review object
+        review_message = review.message
+
+        # Tokenize a message
+        sentences = sent_tokenize(review_message)
+
+        # reject sentences of inappropriate lengths
+        for sentence in sentences:
+            word_count = len(sentence.split())
+            if(word_count > constants.min_words_in_sentence and word_count < constants.max_words_in_sentence):
+                processed_sentences.append(sentence)
+    return processed_sentences
+
+
+"""
```

Comment:
Documentation of function is written inside the function according to the standards and is whats followed in this repo.